### PR TITLE
fix: refactor configuration to do validate in testing

### PR DESCRIPTION
**Pull request by jakeyheath**

### CODE REVIEW

**Review by alexlokshin-czi — APPROVED** [review]

File: terraform/provider/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 test:
-	TF_ACC=yes HAPPY_API_BASE_URL=https://fake.happy-api.io go test -cover -v ./... -run ^$(name)
+	go test -cover -v ./... -run ^$(name)
```
Comment (jakeyheath) [inline terraform/provider/Makefile:3]: moved this into the tests so they are easier to run. I wasn't a fan of magic environment variables (magic to me at least)

File: terraform/provider/pkg/provider/data_resolved_app_configs_test.go [hunk 2]
```diff
@@ -64,16 +88,13 @@ func TestGetResolvedAppConfigsSucceed(t *testing.T) {
 
 	private, _ := generateRsaKeyPair()
 	pemString := exportRsaPrivateKeyAsPemStr(private)
+	os.Setenv("TF_ACC", "yes")
```
Comment (jakeyheath) [inline terraform/provider/pkg/provider/data_resolved_app_configs_test.go:91]: moved environment variable here. this makes it less configurable

File: terraform/provider/pkg/provider/data_resolved_app_configs_test.go [hunk 3]
```diff
@@ -37,7 +38,30 @@ func exportRsaPrivateKeyAsPemStr(privkey *rsa.PrivateKey) string {
 	return string(privkey_pem)
 }
 
+func stashEnv() []string {
```
Comment (jakeyheath) [inline terraform/provider/pkg/provider/data_resolved_app_configs_test.go:41]: saw this in a lot of provider code online and thought it was a good idea and easier to test and cleanup how environment variables were affecting the providers configuration

File: terraform/provider/pkg/provider/provider_test.go [hunk 4]
```diff
@@ -25,7 +25,11 @@ func TestProvider(t *testing.T) {
 func getTestProviders() (map[string]*schema.Provider, *APIMock) {
 	happyProvider := Provider()
 	apiMock := &APIMock{}
-	happyProvider.ConfigureContextFunc = func(context.Context, *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	happyProvider.ConfigureContextFunc = func(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+		_, err := validateConfiguration(d)
```
Comment (jakeyheath) [inline terraform/provider/pkg/provider/provider_test.go:29]: so that we can test that we have valid configuration in unit tests

**Pull request merged**